src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1250,6 +1250,86 @@ public DateTimeOffset GetOldestWalletCreationTime()
             return this.Wallets.Min(w => w.CreationTime);
         }
 
+        /// <inheritdoc />
+        public HashSet<(uint256, DateTimeOffset)> RemoveTransactionsByIds(string walletName, IEnumerable<uint256> transactionsIds)
+        {
+            Guard.NotNull(transactionsIds, nameof(transactionsIds));
+            Guard.NotEmpty(walletName, nameof(walletName));
+
+            List<uint256> idsToRemove = transactionsIds.ToList();
+            Wallet wallet = this.GetWallet(walletName);
+            IEnumerable<HdAccount> accounts = wallet.GetAccountsByCoinType(this.coinType);
```

Comment:
This code `wallet.GetAccountsByCoinType()`  is inside the lock in other places so I wonder if it needs to be inside the lock as well